internal/pkg/service/buffer/storage/slice.go

```diff
@@ -0,0 +1,52 @@
+package storage
+
+import (
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/local"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/staging"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model/column"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const sliceFilename = "slice"
+
+// Slice represents a file slice prepared in the local storage to be uploaded into the staging storage.
+// Slice is part of the File.
+type Slice struct {
+	SliceKey
+	Retryable
+	Type           FileType         `json:"type" validate:"required"`
+	State          SliceState       `json:"state" validate:"required,oneof=writing closing uploading uploaded imported"`
+	ClosingAt      *utctime.UTCTime `json:"closingAt,omitempty" validate:"excluded_if=State writing,required_if=State closing,required_if=State uploading,required_if=State uploaded,required_if=State imported"`
+	UploadingAt    *utctime.UTCTime `json:"uploadingAt,omitempty" validate:"excluded_if=State writing,excluded_if=State closing,required_if=State uploading,required_if=State uploaded,required_if=State imported"`
+	UploadedAt     *utctime.UTCTime `json:"uploadedAt,omitempty" validate:"excluded_if=State writing,excluded_if=State closing,excluded_if=State uploading,required_if=State uploaded,required_if=State imported"`
+	ImportedAt     *utctime.UTCTime `json:"importedAt,omitempty" validate:"excluded_if=State writing,excluded_if=State closing,excluded_if=State uploading,excluded_if=State uploaded,required_if=State imported"`
+	Columns        column.Columns   `json:"columns" validate:"required,min=1"`
```

Comment:
Model schvalne nie je normalizovany, ... mapping `Columns` je vo `File` ale aj v `Slice`.
Takto sme to mali aj doteraz, len poznamka bokom ...

Ked otvorime subor/slice tak mapping je uz immutable, ... a v kode je vyhodne predavat si iba `Slice`, ktora ma vsetky informacie ako aj `File, Export,Receiver` ... a ine nadradene objekty.